helix-lock/src/main/java/org/apache/helix/lock/helix/HelixLockScope.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.helix.lock.helix;
+
+import java.util.List;
+
+import org.apache.helix.lock.LockScope;
+import org.apache.helix.util.StringTemplate;
+
+
+/**
+ *  Defines the various scopes of Helix locks, and how they are represented on Zookeeper
+ */
+public class HelixLockScope implements LockScope {
+
+  /**
+   * Define various properties of Helix lock, and associate them with the number of arguments required for getting znode path
+   */
+  public enum LockScopeProperty {
+
+    CLUSTER(1),
+
+    PARTICIPANT(2),
+
+    RESOURCE(3),
+
+    PARTITION(4);
+
+    //the number of arguments required to generate a full path for the specific scope
+    final int _pathArgNum;
+
+    /**
+     * Initialize a LockScopeProperty
+     * @param pathArgNum the number of arguments required to generate a full path for the specific scope
+    \     */
+    private LockScopeProperty(int pathArgNum) {
+      _pathArgNum = pathArgNum;
+    }
+
+    /**
+     * Get the number of template arguments required to generate a full path
+     * @return number of template arguments in the path
+     */
+    public int getPathArgNum() {
+      return _pathArgNum;
+    }
+
+    /**
+     * Get the position of this argument from the input that used to generate the scope
+     * @return the number of position of value for this property in the list of keys input
+     */
+    public int getArgPos() {
+      return _pathArgNum - 1;
+    }
+  }
+
+  /**
+   * string templates to generate znode path
+   */
+  private static final StringTemplate template = new StringTemplate();
+
+  static {
+    template.addEntry(LockScopeProperty.CLUSTER, 1, "/{clusterName}/LOCK");
```

Comment:
This will be a shared path with the other scope types. I think it might cause some conflicts.

Reply:
Changed it to the same format as path for cluster in HelixConfigScope, should not have conflicts now.